<a href="https://colab.research.google.com/github/nrsur-catalog/NRSurCat-1/blob/gh-pages/_sources/events/GW190521_030229.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open in Colab"/></a>

In [1]:
%load_ext autoreload
%autoreload 2
%matplotlib inline

import pandas as pd

pd.set_option("display.max_rows", None, "display.max_columns", None)

In [2]:
! pip install nrsur_catalog

# GW190521_030229

Below are some plots for GW190521_030229 from the NRSurrogate Catalog.

In [3]:
from nrsur_catalog import NRsurResult

nrsur_result = NRsurResult.load("GW190521_030229", cache_dir=".nrsur_catalog_cache")
# you can specify a `cache_dir`: folder where data will be downloaded

## Summary

In [4]:
nrsur_result.summary()

Posterior 90% CI  \
Parameter                                                                     
$\chi_1$                                           ${0.66}_{-0.43}^{+0.24}$   
$\chi_2$                                           ${0.69}_{-0.42}^{+0.23}$   
$\phi$                                             ${5.05}_{-2.89}^{+0.30}$   
$\mathcal{M}_{c}^{\rm det}\ [M_{\odot}]$         ${114.27}_{-7.85}^{+8.02}$   
$t_c\ [s]$                                ${1242442967.44}_{-0.04}^{+0.00}$   
$d_L\ [Mpc]$                              ${4545.93}_{-1668.08}^{+1594.27}$   
$q$                                                ${0.78}_{-0.21}^{+0.15}$   
$\phi$                                             ${3.73}_{-2.59}^{+1.48}$   
$\phi_{12}$                                        ${3.32}_{-2.33}^{+2.04}$   
$\phi_{\mathrm{JL}}$                               ${3.72}_{-2.58}^{+1.21}$   
$\psi$                                             ${1.76}_{-0.94}^{+0.75}$   
$\theta_{\mathrm{JN}}$                             ${0.88}_{-0.45}^{+1.70}$   
$\theta_1$                                         ${1.58}_{-0.57}^{+0.56}$   
$\theta_2$                                         ${1.56}_{-0.61}^{+0.55}$   
${M}^{\rm det}\ [M_{\odot}]$                   ${266.44}_{-13.36}^{+18.05}$   
$\theta$                                           ${1.77}_{-0.05}^{+0.05}$   
$m_1^{\rm src}\ [M_{\odot}]$                    ${89.27}_{-10.89}^{+13.44}$   
$m_2^{\rm src}\ [M_{\odot}]$                    ${67.05}_{-13.30}^{+14.46}$   
$\chi_{\mathrm{eff}}$                              ${0.00}_{-0.18}^{+0.20}$   
$\chi_{\mathrm{p}}$                                ${0.67}_{-0.25}^{+0.19}$   
$\alpha$                                           ${3.32}_{-0.12}^{+2.60}$   
$\delta$                                           ${0.48}_{-1.38}^{+0.20}$   
$M_f^{\rm src}\ [M_{\odot}]$                   ${253.89}_{-11.77}^{+15.41}$   
$\chi_f$                                           ${0.70}_{-0.06}^{+0.06}$   
$v_f\ [{\rm km/s}]$                          ${708.28}_{-449.99}^{+790.20}$   

                                                                                    Prior  
Parameter                                                                                  
$\chi_1$                                                       $\text{Uniform} [0, 0.99]$  
$\chi_2$                                                       $\text{Uniform} [0, 0.99]$  
$\phi$                                                         $\text{Uniform} [0, 6.28]$  
$\mathcal{M}_{c}^{\rm det}\ [M_{\odot}]$  $\text{UniformInComponentsChirpMass} [12, 400]$  
$t_c\ [s]$                                $\text{Uniform} [1242442967.36, 1242442967.56]$  
$d_L\ [Mpc]$                                 $\text{UniformSourceFrame} [100.0, 10000.0]$  
$q$                                       $\text{UniformInComponentsMassRatio} [0.17, 1]$  
$\phi$                                                         $\text{Uniform} [0, 6.28]$  
$\phi_{12}$                                                    $\text{Uniform} [0, 6.28]$  
$\phi_{\mathrm{JL}}$                                           $\text{Uniform} [0, 6.28]$  
$\psi$                                                         $\text{Uniform} [0, 3.14]$  
$\theta_{\mathrm{JN}}$                                            $\text{Sine} [0, 3.14]$  
$\theta_1$                                                        $\text{Sine} [0, 3.14]$  
$\theta_2$                                                        $\text{Sine} [0, 3.14]$  
${M}^{\rm det}\ [M_{\odot}]$                                $\text{Constraint} [60, 400]$  
$\theta$                                                          $\text{Sine} [0, 3.14]$  
$m_1^{\rm src}\ [M_{\odot}]$                                                          $-$  
$m_2^{\rm src}\ [M_{\odot}]$                                                          $-$  
$\chi_{\mathrm{eff}}$                                                                



| Parameter                                | Posterior 90% CI                  | Prior                                           |
|:-----------------------------------------|:----------------------------------|:------------------------------------------------|
| $\chi_1$                                 | ${0.66}_{-0.43}^{+0.24}$          | $\text{Uniform} [0, 0.99]$                      |
| $\chi_2$                                 | ${0.69}_{-0.42}^{+0.23}$          | $\text{Uniform} [0, 0.99]$                      |
| $\phi$                                   | ${5.05}_{-2.89}^{+0.30}$          | $\text{Uniform} [0, 6.28]$                      |
| $\mathcal{M}_{c}^{\rm det}\ [M_{\odot}]$ | ${114.27}_{-7.85}^{+8.02}$        | $\text{UniformInComponentsChirpMass} [12, 400]$ |
| $t_c\ [s]$                               | ${1242442967.44}_{-0.04}^{+0.00}$ | $\text{Uniform} [1242442967.36, 1242442967.56]$ |
| $d_L\ [Mpc]$                             | ${4545.93}_{-1668.08}^{+1594.27}$ | $\text{UniformSourceFrame} [100.0, 10000.0]$    |
| $q$                                      | ${0.78}_{-0.21}^{+0.15}$          | $\text{UniformInComponentsMassRatio} [0.17, 1]$ |
| $\phi$                                   | ${3.73}_{-2.59}^{+1.48}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi_{12}$                              | ${3.32}_{-2.33}^{+2.04}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi_{\mathrm{JL}}$                     | ${3.72}_{-2.58}^{+1.21}$          | $\text{Uniform} [0, 6.28]$                      |
| $\psi$                                   | ${1.76}_{-0.94}^{+0.75}$          | $\text{Uniform} [0, 3.14]$                      |
| $\theta_{\mathrm{JN}}$                   | ${0.88}_{-0.45}^{+1.70}$          | $\text{Sine} [0, 3.14]$                         |
| $\theta_1$                               | ${1.58}_{-0.57}^{+0.56}$          | $\text{Sine} [0, 3.14]$                         |
| $\theta_2$                               | ${1.56}_{-0.61}^{+0.55}$          | $\text{Sine} [0, 3.14]$                         |
| ${M}^{\rm det}\ [M_{\odot}]$             | ${266.44}_{-13.36}^{+18.05}$      | $\text{Constraint} [60, 400]$                   |
| $\theta$                                 | ${1.77}_{-0.05}^{+0.05}$          | $\text{Sine} [0, 3.14]$                         |
| $m_1^{\rm src}\ [M_{\odot}]$             | ${89.27}_{-10.89}^{+13.44}$       | $-$                                             |
| $m_2^{\rm src}\ [M_{\odot}]$             | ${67.05}_{-13.30}^{+14.46}$       | $-$                                             |
| $\chi_{\mathrm{eff}}$                    | ${0.00}_{-0.18}^{+0.20}$          | $-$                                             |
| $\chi_{\mathrm{p}}$                      | ${0.67}_{-0.25}^{+0.19}$          | $-$                                             |
| $\alpha$                                 | ${3.32}_{-0.12}^{+2.60}$          | $-$                                             |
| $\delta$                                 | ${0.48}_{-1.38}^{+0.20}$          | $-$                                             |
| $M_f^{\rm src}\ [M_{\odot}]$             | ${253.89}_{-11.77}^{+15.41}$      | $-$                                             |
| $\chi_f$                                 | ${0.70}_{-0.06}^{+0.06}$          | $-$                                             |
| $v_f\ [{\rm km/s}]$                      | ${708.28}_{-449.99}^{+790.20}$    | $-$                                             |


_Note: For most events the kick inference is dominated by the prior itself, see Figs. 10 and 11 of [Islam et al, 2023](https://arxiv.org/abs/2309.14473)._

Lets make some plots!

In [5]:
# NRSurrogate corner plots

import os

param_sets = dict(
    mass=["mass_1_source", "mass_2_source", "mass_ratio"],
    spin=["a_1", "a_2", "tilt_1", "tilt_2"],
    effective_spin=["mass_ratio", "chi_eff", "chi_p"],
    sky_localisation=["luminosity_distance", "ra", "dec"],
    remnant=["final_mass", "final_spin", "final_kick"],
)
for name, params in param_sets.items():
    fname = f"GW190521_030229_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_corner(params)
        fig.savefig(fname)

    if name == "remnant":
        continue

    # LVK-Comparison plots
    fname = f"GW190521_030229_compare_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_lvk_comparison_corner(params)
        fig.savefig(fname)


## Corner Plots


Corner plots of the NRSurCat-1 posteriors. The plots along the diagonal display the 1D marginalized posteriors, with the median values and the central $1\sigma$ credible regions indicated as text. The contour plots display the joint 2D posteriors at the $1,2,3\sigma$ regions. The LVK posteriors are plotted using the IMRPhenomXPHM samples obtained from the the [GWTC-2.1](https://zenodo.org/record/6513631) and [GWTC-3](https://zenodo.org/record/5546663) zenodo releases. 

### Mass


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190521_030229_mass_corner.png"](GW190521_030229_mass_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190521_030229_compare_mass_corner.png"](GW190521_030229_compare_mass_corner.png)
:::

::::

Posterior corner plots for the component masses, and mass-ratio ($m_1^{\rm src}, m_2^{\rm src}, q$). Component masses are shown in the source frame.



### Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190521_030229_spin_corner.png"](GW190521_030229_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190521_030229_compare_spin_corner.png"](GW190521_030229_compare_spin_corner.png)
:::

::::

Posterior corner plots for the component spin magnitudes ($\chi_1, \chi_2$) and tilt angles ($\theta_1, \theta_2$).


### Effective Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190521_030229_effective_spin_corner.png"](GW190521_030229_effective_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190521_030229_compare_effective_spin_corner.png"](GW190521_030229_compare_effective_spin_corner.png)
:::

::::

Posterior corner plots for the mass-ratio $q$, effective spin $\chi_{\rm eff}$ and the transverse spin precession parameters $\chi_p$.


### Sky-localisation


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190521_030229_sky_localisation_corner.png"](GW190521_030229_sky_localisation_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190521_030229_compare_sky_localisation_corner.png"](GW190521_030229_compare_sky_localisation_corner.png)
:::

::::

Posterior corner plots for the luminosity distance, ra and dec ($d_L, \alpha, \delta$).



### Remnant

!["GW190521_030229_remnant_corner.png"](GW190521_030229_remnant_corner.png)


Posterior corner plots for the remnant mass, remnant spin magnitude and kick ($M_f^{\rm src}, \chi_f, v_f$).

## Waveform posterior-predictive plot

This is a plot of waveforms generated using 1000 random posterior samples from the event's posterior.

In [6]:
fname = f"GW190521_030229_waveform.png"
if not os.path.isfile(fname):
    fig = nrsur_result.plot_signal(outdir=".")

![waveform](GW190521_030229_waveform.png)

## Animations  
Here are some animations of the posterior.

|Spin       | 
|-----------|
| <video width="630" height="315" controls muted loop autoplay><source src="https://nrsur-catalog.github.io/NRSurCat-1-animations-spins/GW190521_030229_spins.mp4" type="video/mp4"> Your browser does not support the video tag.</video>    |  

Posterior samples for the dimensionless spin vectors ($\chi_1, \chi_2$) 
of the component black holes. Each purple marker indicates a posterior sample; 
an arrow drawn from the origin to the marker would show the spin vector. 
The outer radii of the spheres correspond to the maximum spin magnitude of 1. 
The spins are shown in the "wave frame" defined at a reference frequency of 20 Hz. 
The x-axis (orange) and y-axis (green) are shown as arrows near the origin; 
the x-y plane is orthogonal to the orbital angular momentum direction. 
The color reflects posterior probability density.

|Remnant       | 
|-----------|
| <video width="630" height="315" controls muted loop autoplay><source src="https://nrsur-catalog.github.io/NRSurCat-1-animations-remnant/GW190521_030229_remnant.mp4" type="video/mp4"> Your browser does not support the video tag.</video>    | 

Posterior samples for the dimensionless spin vector $\chi_f$ and the recoil kick velocity 
$v_f$, (units of km/s). Each purple marker indicates a posterior sample; 
an arrow drawn from the origin to the marker would show the spin or kick vector. 
For the spin, the outer radii of the spheres correspond to the maximum spin magnitude of 1. 
For the kick, the outer radius of the sphere corresponds to a kick magnitude of 2500 km/s.
The remnant spin and kick are shown in the "wave frame" defined at a 
reference time of -100 $M^{\rm det}\ [M_{\odot}]$ before the peak waveform amplitude,
where $M^{\rm det}$ is the detector frame total mass.


## Analysis configs
Below are the configs used for the analysis of this job.


In [7]:
nrsur_result.print_configs()


Tmax:                               10000
accounting:                         None
accounting_user:                    None
adapt:                              False
allow_tape:                         True
analysis_executable:                None
analysis_executable_parser:         None
autocorr_c:                         5.0
autocorr_tol:                       50.0
bilby_zero_likelihood_mode:         False
burn_in_nact:                       50.0
calibration_model:                  CubicSpline
calibration_prior_boundary:         reflective
catch_waveform_errors:              True
channel_dict:                       {H1:DCS-CALIB_STRAIN_CLEAN_SUB60HZ_C01, L1:DCS-CALIB_STRAIN_CLEAN_SUB60HZ_C01, V1:Hrec_hoft_16384Hz}
check_point_deltaT:                 3600
clean:                              False
coherence_test:                     False
conda_env:                          None
condor_job_priority:                0
conversion_function:                None
create_summary:              

## Comments

This has been made with `nrsur_catalog` version 0.0.5.
If you used this data, please [cite this work](citation_section).

Leave a comment in this [issue](https://github.com/nrsur-catalog/NRSurCat-1/issues/new?title=GW190521_030229).